plugins/inputs/execd/execd_unix.go

```diff
@@ -14,6 +14,11 @@ func (e *Execd) Gather(acc telegraf.Accumulator) error {
 	if e.cmd == nil || e.cmd.Process == nil {
 		return nil
 	}
+	select {
+	case <-e.ctx.Done():
+		return nil
+	default:
+	}
```

Comment:
This can never happen as a type invariant, Gather won't be called after Stop() completes.